talk/basicconcepts.tex

```diff
@@ -425,6 +425,61 @@
   \end{exampleblock}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitlegb{Scope and lifetime of resources}
+  \begin{block}{Resource life time}
+    \begin{itemize}
+      \item Resources are allocated when declared
+      \item Resources are freed at the end of a scope
+      \item Good practice: initialise resources when allocating!
+    \end{itemize}
+  \end{block}
+  \begin{multicols}{2}
+    \begin{overprint}[\columnwidth]
+    \onslide<1>
+    \inputminted[linenos,highlightlines={1}]{cpp}{codeSnippets/scopes.cpp}
+    \onslide<2>
+    \inputminted[linenos,highlightlines={3}]{cpp}{codeSnippets/scopes.cpp}
+    \onslide<3>
+    \inputminted[linenos,highlightlines={4}]{cpp}{codeSnippets/scopes.cpp}
+    \onslide<4>
+    \inputminted[linenos,highlightlines={7}]{cpp}{codeSnippets/scopes.cpp}
+    \end{overprint}
```

Comment:
[Restating what I said in another comment as github seem to have lost it...]
I would prefer avoiding an external file for such small code. According to the documentation highlightlines is supported by regular `begin{minted}` so the only problem is to avoid repeating the code for each different highlighting. I thought it would be as trivial as storing it in a variable, but does not seem so straightforward. If you have any idea...

Reply:
I tried to store it somewhere, but didn't come up with anything that works ...
This is probably due to how pygments extracts the source code from latex (you cannot expand anything inside the code block).

Reply:
It was a rather long fight, but I've managed to make something nice :-) See !6. If you can review it and confirm it works for you, it would be great !